File: benchmarks/Java/src/DeltaBlue.java
/*
 * This benchmark is derived from Mario Wolczko's Java and Smalltalk version of
 * DeltaBlue.
 * 
 * It is modified to use the SOM class library and Java 8 features.
 * License details:
 *   http://web.archive.org/web/20050825101121/http://www.sunlabs.com/people/mario/java_benchmarking/index.html
 */
import deltablue.Planner;

public class DeltaBlue extends Benchmark {

  @Override
  public boolean innerBenchmarkLoop(final int innerIterations) {
    Planner.chainTest(innerIterations);
    Planner.projectionTest(innerIterations);
    return true;
  }

  @Override
  public Object benchmark() {
    throw new UnsupportedOperationException("should never be reached");
  }

  @Override
  public boolean verifyResult(final Object result) {
    throw new UnsupportedOperationException("should never be reached");
  }
}

File: benchmarks/C++HJ/DeltaBlue.cpp
#include "Benchmark.cpp"
#include "som/Error.cpp"
#include "deltablue/Planner.h"
#include <iostream>

using namespace std;

namespace deltablue {
    class DeltaBlue : public Benchmark {
        public:
            bool innerBenchmarkLoop(int innerIterations) override {
                Planner::chainTest(innerIterations);
                Planner::projectionTest(innerIterations);
                return true;
            } 

            any benchmark() override {
                throw Error("should never be reached");
            }
    
            bool verifyResult(any result) override {
                throw Error("should never be reached");
            }
    };
}   